### PR TITLE
add consumer and update user logic

### PR DESCRIPTION
**Pull request by BelehovEgor**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/635

### CODE REVIEW

File: src/CompanyService.Data.Provider.MsSql.Ef/Migrations/20210601145730_AddDbPositionUser.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            migrationBuilder.CreateTable(
+                name: DbPositionUser.TableName,
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    PositionId = table.Column<Guid>(nullable: false),
+                    UserId = table.Column<Guid>(nullable: false),
+                    IsActive = table.Column<bool>(nullable: false),
+                    StartTime = table.Column<DateTime>(nullable: false),
+                    EndTime = table.Column<DateTime?>(nullable: true)
```
Comment (djino97) [inline src/CompanyService.Data.Provider.MsSql.Ef/Migrations/20210601145730_AddDbPositionUser.cs:23]: Why this need?

File: src/CompanyService.Data/Interfaces/IDepartmentUserRepository.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using LT.DigitalOffice.CompanyService.Models.Db;
+using LT.DigitalOffice.Kernel.Attributes;
+using System;
+
+namespace LT.DigitalOffice.CompanyService.Data.Interfaces
+{
+    [AutoInject]
+    public interface IDepartmentUserRepository
+    {
+        public bool Add(DbDepartmentUser departmentUser);
```
Comment (djino97) [inline src/CompanyService.Data/Interfaces/IDepartmentUserRepository.cs:10]: remove public and add empty separated lines

File: src/CompanyService.Data/Interfaces/IPositionUserRepository.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using LT.DigitalOffice.CompanyService.Models.Db;
+using LT.DigitalOffice.Kernel.Attributes;
+using System;
+
+namespace LT.DigitalOffice.CompanyService.Data.Interfaces
+{
+    [AutoInject]
+    public interface IPositionUserRepository
+    {
+        public bool Add(DbPositionUser positionUser);
```
Comment (djino97) [inline src/CompanyService.Data/Interfaces/IPositionUserRepository.cs:10]: remove public

File: src/CompanyService.Data/PositionRepository.cs [hunk 4]
```diff
@@ -43,6 +43,20 @@ public List<DbPosition> FindPositions()
...
+            var dbCompanyUser = _provider.PositionUsers
+                .Include(u => u.Position)
+                .FirstOrDefault(pu => pu.UserId == userId);
+
+            if (dbCompanyUser == null)
+            {
+                throw new NotFoundException($"User with id: '{userId}' was not found.");
+            }
+
+            return _provider.Positions.Find(dbCompanyUser.PositionId);
```
Comment (djino97) [inline src/CompanyService.Data/PositionRepository.cs:57]: this look strange, may be simple write `dbCompanyUser .Position`? and why this method not place in PositionUserRepository?

File: test/CompanyService.Broker.UnitTests/ChangeUserDepartmentConsumerTests.cs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+                .Throws(new Exception());
+
+            try
+            {
+                _requestClient = await _harness.ConnectRequestClient<IChangeUserDepartmentRequest>();
+
+                var response = await _requestClient.GetResponse<IOperationResult<bool>>(
+                    IChangeUserDepartmentRequest.CreateObj(userId, departmentId));
+
+                Assert.IsFalse(response.Message.IsSuccess);
```
Comment (djino97) [inline test/CompanyService.Broker.UnitTests/ChangeUserDepartmentConsumerTests.cs:111]: not all asserts checked

**Review by MariyaSin — APPROVED** [review]

File: src/CompanyService.Data.Provider.MsSql.Ef/Migrations/20210601145730_AddDbPositionUser.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/CompanyService.Data.Provider.MsSql.Ef/Migrations/20210601145730_AddDbPositionUser.cs:23]: What? endtime? it was in models of department user + i don't think that it is unnecessary info

File: src/CompanyService.Data/Interfaces/IDepartmentUserRepository.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/CompanyService.Data/Interfaces/IDepartmentUserRepository.cs:10]: fixed

File: src/CompanyService.Data/Interfaces/IPositionUserRepository.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/CompanyService.Data/Interfaces/IPositionUserRepository.cs:10]: fixed

File: src/CompanyService.Data/PositionRepository.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/CompanyService.Data/PositionRepository.cs:57]: it was method which I forgot to delete

File: test/CompanyService.Broker.UnitTests/ChangeUserDepartmentConsumerTests.cs (hunk 5 above)
Comment (BelehovEgor) [reply test/CompanyService.Broker.UnitTests/ChangeUserDepartmentConsumerTests.cs:111]: fixed

**Review by djino97 — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

**Review by TIove — APPROVED** [review]

**Pull request merged**